src/System.Runtime.InteropServices.RuntimeInformation/src/RuntimeInformation.Windows.cs

```diff
@@ -1,13 +1,77 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System.Diagnostics;
+
 namespace System.Runtime.InteropServices
 {
-    public static class RuntimeInformation
+    public static partial class RuntimeInformation
     {
+        private static string s_osDebugName = null;
+        private static Architecture? s_osArch = null;
+
         public static bool IsOSPlatform(OSPlatform osPlatform)
         {
             return OSPlatform.Windows == osPlatform;
         }
+
+        public static string GetOSDebugName()
+        {
+            if (null == s_osDebugName)
+            {
+                string type = Interop.mincore.IsWindowsServer() ? "Server" : "Client";
+                string version = "Unknown";
+
+                if (Interop.mincore.IsWindows10OrGreater())
+                    version = "10";
```

Comment:
Even if we did call GetVersion it would be of little use as it would be the same value all the time as Windows started lying about it and always returns the version for Windows 8, no matter which later version of windows you are running on. Unless you explicitly add a manifest that states you are a app targeting a higher version, at which point it will essentially always return what you have in your manifest.